components/combobox/combobox.jsx

```diff
@@ -1004,8 +1025,11 @@ class Combobox extends React.Component {
 			!this.props.multiple && !isSelected;
 		const multiSelectAndSelectedWasNotClicked =
 			this.props.multiple && !isSelected;
+		const deselectWasClicked = option.id === `${this.getId()}-deselect`;
```

Comment:
The string template 
```
`${this.getId()}-deselect`
```
is used more than once and could be dried with a method.

This would also help prevent errors if the suffix `-deselect` ever needs to change for some reason.

Reply:
Class property added to make it DRYer.